runtime/vm/jvminit.c

```diff
@@ -2006,9 +2006,10 @@ VMInitStages(J9JavaVM *vm, IDATA stage, void* reserved)
 				IDATA argIndex8 = 0;
 				IDATA argIndex9 = 0;
 				BOOLEAN sharedClassDisabled = FALSE;
-				BOOLEAN processPortableSharedCacheFlag = FALSE;
 
 #if (defined(J9VM_ARCH_X86) || defined(J9VM_ARCH_S390) || defined(J9VM_ARCH_POWER) || defined(J9VM_ARCH_AARCH64))
```

Comment:
It doesn't seem the best approach to remove a feature by checking for the platform, in particular when it's needed in multiple places. What motivated the original check for the platforms? I'd prefer `!defined(J9VM_ARCH_RISCV)` or define a feature flag J9VM_PORTABLE_SHARED_CACHE and use it instead.

Reply:
It looks as if `-XX:+PortableSharedCache` support was extended to various platforms in stages, starting with x86 initially. I suppose they all needed to have specific portable processor definitions (e.g., the one for Z https://github.com/eclipse-openj9/openj9/pull/10521).

I can change this to `!defined(J9VM_ARCH_RISCV)` so it builds on RISC-V for now. I'm not sure how to go about adding a new feature flag. That would be cleaner.

Reply:
Look at the buildspecs directory, see j9.flags. An example is opt_jitserver (J9VM_OPT_JITSERVER) or opt_sharedClasses (J9VM_OPT_SHARED_CLASSES), search the directory for one or both of these to see how they are defined. You can define opt_portableSharedCache, add it to the core.features list, and turn it off on platforms where it's not supported (RISC-V).

Reply:
I notice there were 3 places with the list of platforms shown in this PR and you changed one of them. Which is kind of the problem, why we should have a better flag.

Reply:
Unless we think it's worthwhile adding a configuration option (in all the extension repositories), I suggest that rather than adding to `j9.flags`, just define the new macro in `j9cfg.h.ftl` and `j9cfg.h.in` like is done for `J9VM_OPT_MULTI_LAYER_SHARED_CLASS_CACHE`:
```c
#if defined(J9VM_OPT_SHARED_CLASSES) \
    && defined(J9VM_ARCH_AARCH64) || defined(J9VM_ARCH_POWER) || defined(J9VM_ARCH_S390) || defined(J9VM_ARCH_X86))
#define J9VM_PORTABLE_SHARED_CACHE
#endif
```
Even if we do want a configuration option, `j9cfg.h.in` would change.

Reply:
That would be more consistent with existing flags.